### PR TITLE
VideoBackends:Metal: Support multiple compute textures

### PR DESCRIPTION
**Pull request by TellowKrinkle**
#11923 for Metal

### CODE REVIEW

**Comment by iwubcode** [comment]
This looks pretty good to me.  I'll try and give it a more thorough look in the coming days.

@OatmealDome please take a look / test when you have time.  At the moment, this feature isn't usable so we just need to ensure nothing breaks.

File: Source/Core/VideoBackends/Metal/MTLUtil.mm [hunk 1]
```diff
@@ -503,6 +497,23 @@ fragment float4 fbfetch_test(float4 in [[color(0), raster_order_group(0)]]) {
 
   for (auto& binding : resource_bindings)
     compiler.add_msl_resource_binding(binding);
+  if (stage == ShaderStage::Pixel)
+  {
+    for (u32 i = 0; i < VideoCommon::MAX_PIXEL_SHADER_SAMPLERS; i++)  // ps/samp0-N
+      compiler.add_msl_resource_binding(
```
Comment (lioncash) [inline Source/Core/VideoBackends/Metal/MTLUtil.mm:503]: These loops (here and below) should be braced, since their body travels more than one line.

**Review by lioncash — COMMENTED** [review]
Just a general Q, but has any testing by anyone been done? This has been sitting here for quite a while, and I'd like to get it merged

File: Source/Core/VideoBackends/Metal/MTLUtil.mm (hunk 1 above)
Comment (TellowKrinkle) [reply Source/Core/VideoBackends/Metal/MTLUtil.mm:504]: Whoops, they were one-liners before clang-format

**Review by lioncash — APPROVED** [review]
OK, given on one of the previous comment stating that this is currently unusable and just needs to ensure nothing else breaks, I can reasonably trust that.

**Pull request merged**